src/MICore/TerminalLauncher.cs

```diff
@@ -106,7 +95,54 @@ protected override string GetProcessArgs()
                 throw new FileNotFoundException(message);
             }
 
-            return string.Format(CultureInfo.InvariantCulture, "{0} \"{1}\" \"{2}\"", launchScript, _title, _initScript);
+            string setEnvironmentScript = GetSetEnvironmentScript();
+            return string.Format(CultureInfo.InvariantCulture, "{0} \"{1}\" \"{2}{3}\"", launchScript, _title, setEnvironmentScript, _initScript);
```

Comment:
I know setEnvironmentScript will end with a space, but I though there was some sort of mistake because there was no space between `{2}` and `{3}`. Can you add a space there?